tests/test_conn.py

```diff
@@ -0,0 +1,874 @@
+"""Unit test for modules/flowalerts/conn.py"""
+
+from tests.module_factory import ModuleFactory
+import json
+from unittest.mock import MagicMock, Mock
+import pytest
+from ipaddress import ip_address
+
+# dummy params used for testing
+profileid = "profile_192.168.1.1"
+twid = "timewindow1"
+uid = "CAeDWs37BipkfP21u8"
+timestamp = 1635765895.037696
+saddr = "192.168.1.1"
+daddr = "192.168.1.2"
+dst_profileid = f"profile_{daddr}"
+
+
+@pytest.mark.parametrize(
+    "dport, proto, daddr, expected_result",
+    [  # Testcase 1: Protocol is not UDP
+        (1234, "tcp", "192.168.1.0", False),
+        # Testcase 2: Port number is less than 30000
+        (8080, "udp", "192.168.1.0", False),
+        # Testcase 3: It is a P2P connection
+        (30001, "udp", "192.168.1.6", True),
+        # Testcase 4: New IP, becomes P2P after 5 connections
+        (30001, "udp", "192.168.1.7", True),
+        # Testcase 5: 6th connection to the same IP,
+        # different port > 30000
+        (30002, "udp", "192.168.1.6", True),
+        # Testcase 6: Existing IP, but already marked as P2P
+        (30003, "udp", "192.168.1.1", True),
+    ],
+)
+def test_is_p2p(mock_db, dport, proto, daddr, expected_result):
+    mock_conn = MagicMock()
+    mock_conn.p2p_daddrs = {
+        "192.168.1.1": 1,
+        "192.168.1.2": 1,
+        "192.168.1.3": 1,
+        "192.168.1.4": 1,
+        "192.168.1.5": 1,
+    }
+    conn = ModuleFactory().create_conn_analyzer_obj(mock_db)
+    conn.p2p_daddrs = mock_conn.p2p_daddrs
+    assert conn.is_p2p(dport, proto, daddr) == expected_result
```

Comment:
in the cases where the expected result is True, it's always hitting this condition if len(self.p2p_daddrs) >= 5: and returnig True, we need to also test if self.p2p_daddrs[daddr] >= 6. and p2p_daddrs shouldn't be the same in all of them i guess, right?